core/types/block.libevm.go

```diff
@@ -27,40 +28,50 @@ import (
 // HeaderHooks are required for all types registered with [RegisterExtras] for
 // [Header] payloads.
 type HeaderHooks interface {
+	MarshalJSON(*Header) ([]byte, error) //nolint:govet // Type-specific override hook
+	UnmarshalJSON(*Header, []byte) error //nolint:govet
 	EncodeRLP(*Header, io.Writer) error
 	DecodeRLP(*Header, *rlp.Stream) error
 }
 
+// hooks returns the Header's registered HeaderHooks, if any, otherwise a
```

Comment:
nit maybe use moaaarrree brackets? Up for debate hence the nit prefix

```suggestion
// hooks returns the [Header]'s registered [HeaderHooks], if any, otherwise a
```

Reply:
Both of those types are immediately available in the signature, which is why I didn't add them. Granted this isn't an exported comment so even the ones around `NOOPHeaderHooks` are fairly meaningless but I did it as there's no way they'll be added in the event that this gets exported later.

Reply:
Thanks for the explanation!